src/ldp.rs

```diff
@@ -4,8 +4,12 @@ use std::convert::TryFrom;
 use std::str::FromStr;
 
 use async_trait::async_trait;
+use cached::{proc_macro::cached, TimedSizedCache};
 use chrono::prelude::*;
 
+const CACHE_SIZE: usize = 100; // number of items
+const CACHE_TTL: u64 = 60; // in seconds
```

Comment:
Is 60 second expiration the only cache invalidation mechanism? What about implementations where the VM changes in under that? I think the user should have to explicitly add caching into the options if they want it, and by default `ssi` should not cache. Perhaps we can expose this in that fashion?

Reply:
It definitely needs to be configurable, I'm just on the fence on how to do it. I see four options:
- env var (easiest to implement but not flexible);
- have some kind of public mutable global variable;
- add a parameter to each individual resolver; or
- put those public function behind a kind of `SSIClient` struct which would have a builder and whatnot.

I would be in favour of the last one but I don't have a strong opinion.

As for the default, I think we should follow what our customers would expect. We can always limit it to 1 or 2 seconds TTL, which would limit the number of dereferencing to issue a VC for example.

Reply:
Indeed, this is tricky. I think we should err on the side of TLL=0 and making it a compile-time configurable option for users who truly care about caching, or just omit these changes entirely and tell people to use a webcache if they are truly squeezing for performance. I just think the added inefficiency is definitely worth eliminating the 1% case where someone can't figure out why their quickly-changing DID document is only breaking some of the time.